src/ucs/datastruct/bitmap.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_BITMAP_H_
+#define UCS_BITMAP_H_
+
+#include <ucs/sys/compiler_def.h>
+
+BEGIN_C_DECLS
+
+#define _UCS_BITMAP_BITS_IN_WORD (64)
+#define _UCS_BITMAP_WORD_INDEX(_bit_index) ((_bit_index) / _UCS_BITMAP_BITS_IN_WORD )
+#define _UCS_BITMAP_BIT_IN_WORD_INDEX(_bit_index) ((_bit_index) % _UCS_BITMAP_BITS_IN_WORD)
+
+/*
+ * Represents an n-bit bitmap, where n is a multiple of 64, by using an array
+ * of 64-bit unsigned long integers. Memory is allocated on stack
+ * 
+ * @param _bits Bits number in the bitmap
+ */
+#define _UCS_BITMAP_DECLARE_TYPE(_bits) \
+    typedef struct { \
+        uint64_t bits[_UCS_BITMAP_WORD_INDEX(_bits)]; \
+    } ucs_bitmap_t(_bits);
+
+/**
+ * Expands to bitmap type definition
+ *
+ * @param _bits Bits number (as passed to UCS_BITMAP_DECLARE_TYPE)
+ *
+ * Example:
+ *
+ * @code{.c}
+ * ucs_bitmap_t(64) my_bitmap;
+ * @endcode
+ */
+#define ucs_bitmap_t(_bits) \
+    _UCS_BITMAP_IDENTIFIER(_bits, _t)
+
+#define _UCS_BITMAP_IDENTIFIER(_name, _suffix) \
+    UCS_PP_TOKENPASTE(ucs_bitmap_, UCS_PP_TOKENPASTE(_name, _suffix))
+
+_UCS_BITMAP_DECLARE_TYPE(64)
+_UCS_BITMAP_DECLARE_TYPE(128)
```

Comment:
i'd declare 128 and 256 instead, no need for this abstraction for 64 bits

Reply:
@brminich , added 256-bit type (though not sure why - if it is not needed at the moment).
Regarding 64-bit - I'd suggest that we do put it here and use it, so that bitmaps across the code are consistent (and can be easily switched from 64 to 128 for example) - anyway there's not overhead in using this struct in the 64-bit case vs. a regular `uint64_t`. What do you think?

Reply:
ok